test/system_test.py

```diff
@@ -175,7 +179,20 @@ def _check_dcm_tags_from_zip(
             else:
                 assert private_tag.value == TestFtpsUpload.project_slug
         # check the basic info about the instances exactly matches
-        assert actual_instances == expected_instances
+        if len(actual_instances) == 2:
+            assert (
+                actual_instances
+                == expected_studies["1.3.46.670589.11.38023.5.0.14068.2023012517090160001"][
+                    "instances"
+                ]
+            )
+        elif len(actual_instances) == 1:
+            assert (
+                actual_instances
+                == expected_studies["1.3.46.670589.11.38023.5.0.14068.2023012517090160002"][
+                    "instances"
+                ]
+            )
```

Comment:
Why is the number of expected instances variable? I'm assuming this is something to do with the unpredictability of the UID, but then why is there a UID specified here at all if the actual value will never match it? Could we maybe use the procedure_occurrence_id as a fixed ID to identify the right one to compare against?